[libkvifile docs] $file.localdir should document $escape due to popupadd.ExtPopup usecase slash stripping

the-j0k3r opened the pull request:
See discussion in #2295

[//]: # (If your pull request is a fix to an open issue please add fixes #9999 to the commit comments.)
[//]: # (If your proposal involves GUI improvements, add screenshots of before and after to help visualise the proposal on the fly.)
#### Changes proposed
- add documentation to $file.localdir reference to $escape

[//]: # (If you have write privileges to repository, do label your pull request, else you could insert a mention prefixed with @GitHub-Username below.)

the-j0k3r committed:
[libkvifile docs] $file.localdir should document $escape due to popup.addExtPopup usecase slash stripping

See discussion in #2295

the-j0k3r committed:
fix typo: missing brackets

[ci skip]

wodim commented on src/modules/file/libkvifile.cpp:
```diff
@@ -1216,19 +1216,30 @@ static bool file_kvs_cmd_writeLines(KviKvsModuleCommandCall * c)
 		[example]
 			echo KVIrc looks for pictures in $file.localdir(pics)
 			echo panic.png would be translated to $file.localdir(pics/panic.png)
+			echo panic.png would be translated to $escape($file.localdir(pics/panic.png))
```
Why both lines, without `$escape` and with it?

the-j0k3r replied:
I dont know, maybe because you dont need to use with in every case.

wodim replied:
You *need* to use it in every case, unless you want to check if you're on Windows or on Linux before adding a popup menu, which is something you should not do.

The best approach once again is to use `\$` because that way you can move your configuration folder between OSes and have it working, but otherwise `$escape` is a must.

the-j0k3r replied:
Thats not what I found at all and I and others have tested this,

YOU dont need to use it in every case even if OS's are different, and when adding a popup you dont need to check either which docs say clearly what the exception is.

what you suggest is broken in some cases, and I wont add more broken docs.
feel free to close.

Stanzilla replied:
That sounds like a un1versal solution.

the-j0k3r replied:
> That sounds like a un1versal solution.

This is the second time youre making some implications that are false.
I know the person know here as un1versal, he was a very good friend of my father.

Not that, I owe you or anyone any explanations, but un1versal was a good person and did nothing afaik but help. He's passed away now for some time and so has my father, and unless you have something against me (which escapes me because I dont know you or you me), I have nothing else to say.

Stanzilla replied:
I don't have anything against you, I just don't believe the story.

the-j0k3r replied:
I have nothing to prove to you or anyone and have done nothing to you or this project but try to help a friend of my father with some of his projects, your implications are already a waste of time.

Your attitude here of making petty reports to Github are a waste of my time and hinder contributions. If you dont want my contributions just close them,

Stanzilla replied:
I didn't report anything to GitHub and let's leave it at, we would not have a problem if un1versal returned, no need for games.

the-j0k3r replied:
Tread carefully, he was a friend of my fathers, perhaps his best and oldest friend, dont disrespect the dead by making false implications.
I have great respect for the man he was and what he stood for, hes passed away long ago so let that rest already.

the-j0k3r committed:
examples only with $escape

this could possibly be further improved by someone dedicated to the task.
the point here is to had mention of the usage and some relevant text to cure the non mention of this anywhere.

the-j0k3r commented:
@wodim, is that better? Im not adding ```\$escape``` mention because as test showed it breaks popup editor icon not showing for one. this solution works afaik everywhere even if not strictly needed in all cases.

ltc-data commented:
@the-j0k3r dont waste your time here, the people in charge here dont deserve your or my or anyones help.

fyi, turn on your phone.

ltc-data commented:
Your turn. Im out of this pit.
